### PR TITLE
Simplify samples

### PR DESCRIPTION
**Pull request by alexshtin**
Simplified all samples in `cmd/samples` besides `recipes` sub-dir. Moved them on top level.

### CODE REVIEW

File: recovery/worker/main.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	workflowWorker.RegisterWorkflowWithOptions(recovery.RecoverWorkflow, workflow.RegisterOptions{Name: "RecoverWorkflow"})
+	workflowWorker.RegisterWorkflowWithOptions(recovery.TripWorkflow, workflow.RegisterOptions{Name: "TripWorkflow"})
+	workflowWorker.RegisterActivity(recovery.ListOpenExecutions)
+
+	err = workflowWorker.Start()
+	if err != nil {
+		logger.Fatal("Unable to start worker", zap.Error(err))
+	}
+
+	hostWorker := worker.New(c, recovery.HostID, worker.Options{
```
Comment (alexshtin) [inline recovery/worker/main.go]: This seems not working correctly. And I don't understand how worker for random task list (`HostID` is random) can be used.

Comment (alexshtin) [reply recovery/worker/main.go]: @mfateev can you try this sample after I merge it?

File: cmd/samples/common/sample_helper.go [hunk 2]
```diff
@@ -117,14 +117,10 @@ func (h *SampleHelper) SignalWithStartWorkflowWithCtx(ctx context.Context, workf
...
 
 // StartWorker starts workflow worker and activity worker based on configured options.
-func (h *SampleHelper) StartWorker(domainName, groupName string, options worker.Options) worker.Worker {
-	worker, err := worker.New(domainName, groupName, options)
-	if err != nil {
-		h.Logger.Error("Failed to create worker.", zap.Error(err))
-		panic("Failed to start workers")
-	}
+func (h *SampleHelper) StartWorker(client client.Client, groupName string, options worker.Options) worker.Worker {
+	worker := worker.New(client, groupName, options)
```
Comment (mfateev) [inline cmd/samples/common/sample_helper.go]: Wy do we still have this helper?

File: recovery/worker/main.go (hunk 1 above)
Comment (alexshtin) [reply recovery/worker/main.go]: I think I fixed it.

**Pull request merged**